app/webpacker/components/Panel/pages/CompetitorRegistrationPage/index.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+import {
+  Form, Header, Message, Label, Table, Loader,
+} from 'semantic-ui-react';
+import { useQuery } from '@tanstack/react-query';
+import { IdWcaSearch } from '../../../SearchWidget/WcaSearch';
+import SEARCH_MODELS from '../../../SearchWidget/SearchModel';
+import { fetchJsonOrError } from '../../../../lib/requests/fetchWithAuthenticityToken';
+import { viewUrls } from '../../../../lib/requests/routes.js.erb';
+
+async function getCompetitions({ wcaId }) {
+  const { data } = await fetchJsonOrError(
+    viewUrls.persons.registrations(wcaId),
+  );
+  return data || {};
+}
+
+function CompetitorRegistrationPage() {
+  const [formValues, setFormValues] = useState({});
+
+  const { wcaId } = formValues;
+
+  const {
+    data: competitionsList,
+    isFetching: competitionsFetching,
+  } = useQuery({
+    queryKey: ['competitor-registration-competitions', wcaId],
+    queryFn: () => getCompetitions({ wcaId }),
+    enabled: !!wcaId,
+  });
+
+  const handleWcaIdChange = (_, { value: newWcaId }) => setFormValues(
+    (prev) => ({
+      ...prev,
+      wcaId: newWcaId,
+    }),
+  );
```

Comment:
If you have a `setWcaId` from `useState` above, you can use our custom `useInputUpdater` hook to wrap it in a way that SemUI understands.